src/main/kotlin/uk/gov/justice/digital/assessments/services/AssessmentService.kt

```diff
@@ -16,8 +17,12 @@ class AssessmentService (private val assessmentRepository: AssessmentRepository)
     }
 
     fun createNewAssessment(supervisionId : String?) : AssessmentDto {
-        val assessment = assessmentRepository.save(AssessmentEntity(supervisionId = supervisionId, createdDate = LocalDateTime.now()))
-        return AssessmentDto.from(assessment)
+        val existingAssessment = assessmentRepository.findBySupervisionId(supervisionId)
+        if (!existingAssessment.isEmpty())
```

Comment:
We can use  if (existingAssessment.isNotEmpty())

Reply:
Of course.

Reply:
Turns out isNotEmpty is spelled isPresent on an Optional.